src/IO/BufferedOutput.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace PhpSchool\CliMenu\IO;
+
+/**
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class BufferedOutput implements OutputStream
+{
+    private $buffer = '';
+
+    public function write(string $buffer): void
+    {
+        $this->buffer .= $buffer;
+    }
+
+    public function fetch(bool $clean = true) : string
+    {
+        $buffer = $this->buffer;
+
+        if ($clean) {
+            $this->buffer = '';
+        }
+
+        return $buffer;
+    }
+
+    public function __toString() : string
+    {
+        return $this->fetch();
```

Comment:
looks like this would always return an empty string ?

Reply:
Nope - it will return whatever has been written so far and clean the buffer

Reply:
oh my dayz... obvs .. I must have been tired not to see that 😂